Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/GameFileCacheManager.java

```diff
@@ -40,15 +43,20 @@ public static LiveData<GameFile[]> getGameFiles()
     return sGameFiles;
   }
 
-  public static List<GameFile> getGameFilesForPlatform(Platform platform)
+  public static List<GameFile> getGameFilesForPlatform(Context context, Platform platform)
   {
+    boolean legacyPaths = DirectoryInitialization.preferOldFolderPicker(context);
+
     GameFile[] allGames = sGameFiles.getValue();
     ArrayList<GameFile> platformGames = new ArrayList<>();
     for (GameFile game : allGames)
     {
       if (Platform.fromNativeInt(game.getPlatform()) == platform)
       {
-        platformGames.add(game);
+        if (legacyPaths == game.getPath().startsWith("/"))
+        {
+          platformGames.add(game);
+        }
```

Comment:
This is a functionality regression. In master, it's possible to use legacy paths even when `preferOldFolderPicker` returns false.

What's the purpose of this commit? The commit message just says what the commit does, not why.

Reply:
As `preferOldFolderPicker` changes the behavior after installing the VR APK, I want to ensure that there are no games with non-legacy paths which would fail to start.

I will revert this one and do it better.

Reply:
I don't see why those games would fail to start.

Reply:
![photo_2023-04-03_21-55-29](https://user-images.githubusercontent.com/6472545/229613467-bce73b1a-7a31-4338-9464-37c23e79d839.jpg)
As I pass the content URI into another APK the access permission is not transfered. I cannot ask for it because in OpenXR VR context Android UI won't work.

Reply:
https://stackoverflow.com/questions/69751552/how-to-give-to-a-service-permission-to-read-a-uri/69752847#69752847

Reply:
Thanks, it works for URIs passed by "Open file", if I add folder then I need to use the legacy flow.

Reply:
Ah, yes. When you launch a game using a folder you get a "mangled" URI, which doesn't work properly with permissions since it's not a real URI. Try passing the mangled URI to this function before creating the intent: https://github.com/dolphin-emu/dolphin/blob/11281b5cef410976606482fb1591eaf1c88bb2af/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ContentHandler.java#L344

Reply:
Thanks but I am afraid that's not a way to go. It creates from:
`content://com.android.externalstorage.documents/tree/home%3A/Need for Speed - Carbon (USA).wbfs`
this:
`content://com.android.externalstorage.documents/tree/home%3A/document/home%3ANeed%20for%20Speed%20-%20Carbon%20(USA).wbfs`

Reply:
Thank you for the patience, I took a deeper look and figure that out (`takePersistableUriPermission` cannot be called on unmanged URI)

Reply:
I don't think you actually need `takePersistableUriPermission` (or `FLAG_GRANT_PERSISTABLE_URI_PERMISSION`). The VR package will only use the URI for the current emulation session, not forever.